pkg/agent/netpol/network_policy.go

```diff
@@ -28,6 +32,21 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
 		return err
 	}
 
+	// retry backoff to wait for the clusterrolebinding of user "system:k3s-controller"
+	retryBackoff := wait.Backoff{
+		Steps:    6,
+		Duration: 100 * time.Millisecond,
+		Factor:   3.0,
+		Cap:      30 * time.Second,
+	}
+	retryErr := retry.OnError(retryBackoff, errors.IsForbidden, func() error {
+		_, err := client.NetworkingV1().NetworkPolicies("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
Bailout from retry if the context is cancelled.
```suggestion
		_, err := client.NetworkingV1().NetworkPolicies("").List(ctx, metav1.ListOptions{})
```

Reply:
@dweomer good catch, thanks.